client/allocrunner/taskrunner/vault_hook.go

```diff
@@ -364,6 +389,55 @@ func (h *vaultHook) deriveVaultToken() (token string, exit bool) {
 	}
 }
 
+// deriveVaultTokenJWT returns a Vault ACL token using JWT auth login.
+func (h *vaultHook) deriveVaultTokenJWT() (string, error) {
+	// Retrieve signed identity.
+	signed, err := h.widmgr.Get(structs.WIHandle{
+		IdentityName:       h.widName,
+		WorkloadIdentifier: h.task.Name,
+		WorkloadType:       structs.WorkloadTypeTask,
+	})
+	if err != nil {
+		return "", structs.NewRecoverableError(
+			fmt.Errorf("failed to retrieve signed workload identity: %w", err),
+			true,
+		)
+	}
+	if signed == nil {
+		return "", structs.NewRecoverableError(
+			errors.New("no signed workload identity available"),
+			true,
+		)
```

Comment:
In theory the `WIDManager`may not have the identities because of how they are only fetched when `widmgr.Run()` is called, but in practice that doesn't seem like it would be currently possible as it's created and started at the alloc level.

I can change it to an unrecoverable error.